Framework/Geometry/src/Instrument.cpp

```diff
@@ -1306,6 +1239,86 @@ void Instrument::setDetectorInfo(
     m_map_nonconst->setDetectorInfo(detectorInfo);
   m_detectorInfo = std::move(detectorInfo);
 }
+namespace Conversion {
+double calcConversion(const double l1, const Kernel::V3D &beamline,
+                      const double beamline_norm, const Kernel::V3D &samplePos,
+                      const Kernel::V3D &detPos, const double offset) {
+  if (offset <=
+      -1.) // not physically possible, means result is negative d-spacing
+  {
+    std::stringstream msg;
+    msg << "Encountered offset of " << offset
+        << " which converts data to negative d-spacing\n";
+    throw std::logic_error(msg.str());
+  }
 
+  // Now detPos will be set with respect to samplePos
+  Kernel::V3D relDetPos = detPos - samplePos;
+  // 0.5*cos(2theta)
+  double l2 = relDetPos.norm();
+  double halfcosTwoTheta =
+      relDetPos.scalar_prod(beamline) / (l2 * beamline_norm);
+  // This is sin(theta)
+  double sinTheta = sqrt(0.5 - halfcosTwoTheta);
```

Comment:
I would not have minded keeping it (much), but since you moved/renamed it I'd prefer cleaning it up.